### ISSUE TITLE
Dataset setup (e.g., Train, val, test splits)

### ISSUE THREAD

**Issue by seoyoung1215** [facebookresearch/vggt#268]
Hi @jytime,

Thanks for your work! 

I had a few questions regarding the dataset setup for VGGT. Specifically:
1. Could you clarify what train/validation/test splits were used for the Virtual KITTI, Mapillary Metropolis, and MVS-Synth datasets?
2. If there was a general heuristic or protocol applied across datasets (e.g., sequence or camera selection), I’d greatly appreciate if you could share that as well.
3. Additionally, would it be possible to share the dataloaders and preprocessing scripts for Mapillary and MVS-Synth? I’m aiming to reproduce your setup as closely as possible, but ran into some blockers. For example, I wasn't able to find details on the specific configurations or sequence selections for each of the datasets used during training. For example, in the case of Mapillary, were certain camera types used? It seems like CAM_EQUIRECTANGULAR might be a composite of others

I noticed that the current VKITTI dataloader does not explicitly define splits, and was wondering if this the same version used during your training?

Thanks so much in advance!

**Comment by jytime** [comment]
1. We use all of these datasets for training only, and did not use them for validation. So just use as much as you can.
2. Can you elaborate more about this?
3. We pick all the possible sequences of these datasets. For Mapillary, we randomly pick a type of camera when loading the data, and only using the figures of the same camera type. For mvs synth, the dataloader from cut3r (https://github.com/CUT3R/CUT3R/blob/main/datasets_preprocess/preprocess_mvs_synth.py) could be a good example.

**Comment by seoyoung1215** [comment]
I see. Thanks for the clarification! I think I missed how you cleanly separated datasets for training and evaluation.

1. If we trying to finetune from the 1B pretrained VGGT checkpoint (even if we modify the model a bit), and use train/val/test sets from the datasets already used to train the pretrained checkpoint, that would be problematic right? 
2. This was a follow up to the 1st question. If you had a general rule in splitting the datasets, but I think we can skip since you said you use all the data for training.
3. Thanks for the lead to the dataloader! For datasets like Mapillary, that already have train/val/test sets, did you also use all of the splits for training? or just the train sets? Also, did you exclude CAM_EQUIRECTANGULAR as it doesn't have a designated depth folder, or did you somehow preprocess to generate depth maps for CAM_EQUIRECTANGULAR images?

**Comment by jytime** [comment]
1. Generally yes.
2. We did not use the evaluation/test set of Co3D and MegaDepth. We also picked a small part of DL3DV for validation. For other datasets, we use them as much as possible.
3. Yes we used all of Mapillary. Please note the link is Mapillary Metropolis: https://www.mapillary.com/dataset/metropolis. If some sequences do not have gt depth, we filter them out.

**Comment by seoyoung1215** [comment]
Thanks for the clarification! 
Do you also have a recommendation on an example dataloader I could reference for Mapillary Metropolis? There isn't much information on the format of the dataset in the link, so I was a bit lost.

**Comment by jytime** [comment]
Their data format is very straightforward. Once you downloaded the dataset, it should be clear. I cannot provide a file as I have reorganized the data format to our own one.

**Comment by seoyoung1215** [comment]
Ok! Thanks for your comments :)
